Purchases/Public/RCPurchases.m

```diff
@@ -655,6 +655,13 @@ - (void)checkTrialOrIntroductoryPriceEligibility:(NSArray<NSString *> *)productI
                         }];
                     }
                 }];
+            } else {
+                [self.backend getIntroEligibilityForAppUserID:self.appUserID
+                                                  receiptData:data
+                                           productIdentifiers:productIdentifiers
+                                                   completion:^(NSDictionary<NSString *, RCIntroEligibility *> *_Nonnull result) {
+                                                       CALL_IF_SET_ON_MAIN_THREAD(receiveEligibility, result);
+                                                   }];
```

Comment:
These lines is repeated three times in this function. I would extract out of the if-elses and use `return` in the `if` statements.

Reply:
Nevermind, I just noticed they are in completion blocks